continuous_integration/run_tests.sh

```diff
@@ -28,9 +28,7 @@ if [[ "$SKLEARN_TESTS" != "true" ]]; then
 else
     # Install the nightly build of scikit-learn and test against the installed
     # development version of joblib.
-    # TODO: unpin pip once either https://github.com/pypa/pip/issues/10825
-    # accepts invalid HTML or Anaconda is fixed.
-    conda install -y -c conda-forge cython pillow numpy scipy "pip<22"
```

Comment:
The impact of not installing pillow is to skip a few scikit-learn tests (mostly related to a few datasets like fetch_lfw). I don't think we care that much.